src/controllers/EventsController.php

```diff
@@ -68,9 +68,21 @@ public function getAction($request, $db) {
                 $params = array();
 
                 // collection type filter
-                $filters = array("hot", "upcoming", "past", "cfp");
+                $filters = array("hot", "upcoming", "past", "cfp", "pending");
                 if(isset($request->parameters['filter']) && in_array($request->parameters['filter'], $filters)) {
                     $params["filter"] = $request->parameters['filter'];
+
+                    // for pending events we need a logged in user with the correct permissions
+                    if ($params["filter"] == 'pending') {
+                        if (!isset($request->user_id)) {
+                            throw new Exception("You must be logged in to view pending events", 400);
```

Comment:
As described in [RFC5849](https://tools.ietf.org/pdf/rfc5849.pdf#page=27) that should work. But it would need some work on the way the exception triggers the response as we'd have to check for the Exception code being 401 or something else. But thats a different issue.